backend/src/main/java/com/htecgroup/skynest/service/impl/BucketServiceImpl.java

```diff
@@ -137,4 +141,24 @@ public BucketResponse editBucket(
     actionService.recordAction(Collections.singleton(savedBucketEntity), ActionType.EDIT);
     return modelMapper.map(savedBucketEntity, BucketResponse.class);
   }
+
+  @Override
+  public FolderFileResponse getBucketContent(UUID bucketId) {
+
+    List<FolderEntity> allFolders =
```

Comment:
I would move the allFolders and corresponding mapper to separate method in folderService and the same for allFiles, that way it can be reusable and the whole logic wont be in one place

Reply:
Great suggestion. Will do it. It will be later used again in FolderContentResponse so it is really better that way

Reply:
something like getAllFiles(bucketId, parentFolderId)?

in the file service